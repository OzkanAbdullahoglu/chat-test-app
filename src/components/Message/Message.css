.list-item-root {
    width: 100%;
    display: flex;
    position: relative;
    box-sizing: border-box;
    text-align: left;
    align-items: center;
    padding: 0 16px;
    justify-content: flex-start;
    text-decoration: none;
}

.list-item-text-root {
    flex: 1 1 auto;
    min-width: 0;
}

.list-item-text-primary {
    background-color: #fff;
    border-radius: 8px;
    padding: 4px 8px;
    display: flex;
    float: left;
}

.message-out {
    float: right;
    background-color: #beffd9;
}

.time-inline {
    float: right;
    font-size: 0.675rem;
    align-self: center;
    padding: 0 5px 0 8px;
}

.timestamp {
    background-color: #c8c5c5;
    flex: inherit;
    border-radius: 8px;
}