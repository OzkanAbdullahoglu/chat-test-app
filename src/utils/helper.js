/* eslint-disable linebreak-style */
import React, { useState, useEffect } from 'react';
export const timeConverter = (timeStamp) => {
  if (!timeConverter.cache) {
    timeConverter.cache = {};
  }

  if (!timeConverter.cache[timeStamp]) {
    const timeStampMs = timeStamp * 1000;
    const dateObj = new Date(timeStampMs);
    const dayOfMonth = dateObj.getDate() < 10 ? `0${dateObj.getDate()}` : dateObj.getDate();
    const months = [
      'January', 'February', 'March', 'April', 'May', 'June',
      'July', 'August', 'September', 'October', 'November', 'December',
    ];
    const curMonth = months[dateObj.getMonth()];
    const curYear = dateObj.getFullYear();
    const curMinute = dateObj.getMinutes() < 10 ? `0${dateObj.getMinutes()}` : dateObj.getMinutes();
    const curHour = (dateObj.getHours() < 10 ? `0${dateObj.getHours()}` : dateObj.getHours());

    timeConverter.cache[timeStamp] = {
      date: `${curMonth} ${dayOfMonth},${curYear}`,
      time: `${curHour}:${curMinute}`,
    };
    return timeConverter.cache[timeStamp];
  }
  return timeConverter.cache[timeStamp];
};

export const isToday = (timestamp, date) =>{
  const today = Math.floor(Date.now() / 1000).toString();
  console.log(today);
  return timeConverter(today).date === timeConverter(timestamp).date ?
    'Today' : timeConverter(today).date === timeConverter(timestamp).date - 1 ?
    'Yesterday' : date;
};




function getCurrentViewDims() {
  const { innerWidth: width, innerHeight: height } = window;
  return {
    width,
    height,
  };
}

export default function useCurrentViewDims() {
  const [windowDimensions, setWindowDimensions] = useState(getCurrentViewDims());

  useEffect(() => {
    function handleResize() {
      setWindowDimensions(getCurrentViewDims());
    }

    window.addEventListener('scroll', handleResize);
    return () => window.removeEventListener('scroll', handleResize);
  }, []);

  return windowDimensions;
}

