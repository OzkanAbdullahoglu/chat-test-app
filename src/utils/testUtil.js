/* eslint-disable linebreak-style */
import { createStore, applyMiddleware } from 'redux';
import { middlewares } from '../store';
import rootReducer from '../reducers';

const chat = {
  requestedData: [
    {
      id: 1,
      direction: 'in',
      status: 'read',
      timestamp: '1577834102',
      text: 'Alice was beginning',
    },
  ],
  isPending: false,
  error: '',
  scrollDownEnable: true,
};

const message = {
  messagesEndRef: '',
};

const version = { number: '0.0.1' };

export const initialStates = { chat, message, version };

export const storeFactory = (initalState) => {
  const createStoreWithMiddleware = applyMiddleware(...middlewares)(createStore);
  return createStoreWithMiddleware(rootReducer, initalState);
};
