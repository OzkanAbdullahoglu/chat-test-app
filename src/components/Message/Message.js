/* eslint-disable linebreak-style */
import React from 'react';
import Icon from '@material-ui/core/Icon';
import PropTypes from 'prop-types';
import './Message.css';
const Message = ({
  pack,
}) => {
  const IconStyle = {
    fontSize: '1rem',
    padding: 0,
    float: 'right',
    alignSelf: 'center',
    color: pack.status === 'read' ? 'blue' : '#777',
  };
  return (
    <div className="list-item-root">
      <div className="list-item-text-root">
        {pack.direction === 'out' ? (
          <div className="list-item-text-primary message-out">
            <span data-test="item-text">{pack.text}</span>
            <span
              className="time-inline"
            >{pack.time}</span>
            <Icon
              style={IconStyle}
              data-test={pack.status === 'sent' ? 'sent-icon' : 'read-icon'}
            >{pack.status === 'sent' ? 'done' : 'done_all'}</Icon>
          </div>
        ) : (
          <div className="list-item-text-primary">
            <span data-test="item-text">{pack.text}</span>
            <span className="time-inline">{pack.time}</span>
          </div>
        )
        }
      </div>
    </div>
  );
};

Message.propTypes = {
  pack: PropTypes.object,
};

export default Message;
