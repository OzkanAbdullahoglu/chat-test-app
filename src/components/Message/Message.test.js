/* eslint-disable linebreak-style */
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Message from './Message';
import { storeFactory } from '../../utils/testUtil';


describe('Message Component', () => {
  const defaultProps = {
    pack: {
      id: 1,
      direction: 'in',
      status: 'read',
      timestamp: '1577834102',
      text: 'Alice was beginning',
      date: 'January 01,2020',
      time: '02:15',
      height: 45,
    },
  };
  const message = shallow(<Message {...defaultProps} />);

  it('renders without crashing', () => {
    expect(toJson(message)).toMatchSnapshot();
  });
  it('renders properly up to the `pack` prop', () => {
    expect(message.find('[className="time-inline"]').text()).toBe(defaultProps.pack.time);
    expect(message.find('[data-test="item-text"]').text()).toBe(defaultProps.pack.text);
  });
  it('renders conditionally up to the `pack.direction` and `pack.status` prop', () => {
    message.setProps({
      pack: {
        id: 1,
        direction: 'out',
        status: 'read',
        timestamp: '1577834102',
        text: 'Alice was beginning',
        date: 'January 01,2020',
        time: '02:15',
        height: 45,
      },
    });
    expect(message.find('[data-test="read-icon"]').exists()).toBe(true);
    const iconStyle = message.find('[data-test="read-icon"]').get(0).props.style;
    expect(iconStyle).toHaveProperty('color', 'blue');
  });
});
