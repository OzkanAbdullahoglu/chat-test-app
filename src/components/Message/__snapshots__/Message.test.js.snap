// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message Component renders without crashing 1`] = `
<div
  className="list-item-root"
>
  <div
    className="list-item-text-root"
  >
    <div
      className="list-item-text-primary"
    >
      <span
        data-test="item-text"
      >
        Alice was beginning
      </span>
      <span
        className="time-inline"
      >
        02:15
      </span>
    </div>
  </div>
</div>
`;
